src/lib.rs

```diff
@@ -1,53 +1,117 @@
+//! Inquirer – Fancy user interaction on the command line
+
+// Termion depends on nighty (for Keys), we need this for the unstable `std::io::CharsError` 
+#![feature(io)]
+
+#![deny(missing_docs,
+    missing_debug_implementations, missing_copy_implementations,
+    trivial_casts, trivial_numeric_casts,
+    unsafe_code,
+    unused_import_braces, unused_qualifications)]
+
 extern crate termion;
+#[macro_use] extern crate quick_error;
 
-use termion::{TermRead, TermWrite, IntoRawMode, color, Style, Key};
 use std::io::{Write, stdout, stdin};
+use std::fmt::Display;
+use termion::{TermRead, TermWrite, IntoRawMode, color, Style, Key};
+
+/// An option the user can choose
+///
+/// (Since the name "Option" is reserved for the well-known type representing
+/// nullability, we are calling this one "Choice".)
+pub trait Choice {
+    /// User visible text
+    type Text: Display;
+    /// Internal value representing this choice
+    type Value;
+
+    /// Get a reference to the text
+    fn text(&self) -> &Self::Text;
+
+    /// Get a reference to the value of this choice
+    fn value(&self) -> &Self::Value;
+}
+
+impl<'a> Choice for &'a str {
+    type Text = &'a str;
+    type Value = &'a str;
+
+    fn text(&self) -> &Self::Text {
+        self
+    }
+
+    fn value(&self) -> &Self::Value {
```

Comment:
Nice example. I think optimising for lazy programmers is a worthy goal! Though, how is writing a match statement for each case in the enum less work than writing a tuple instead of a string for each enum case?

I _just_ commented below (on the PR, not this change), that it might be nice to have an `impl Choice for T where T: Display`. That means, if your color enum implements display (either custom or `#[derive(Display)]`, it'd just work :)